src/api/spec/models/watched_item_spec.rb

```diff
@@ -0,0 +1,185 @@
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["Watchable has already been taken"])
+      end
+    end
+
+    context 'when the user tries to add different packages to the watchlist' do
+      let(:watched_item) { build(:watched_item, :for_packages) }
+
+      before { create(:watched_item, :for_packages) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+
+    context 'when two different users try to watch the same package' do
+      let(:watched_item) { create(:watched_item, :for_packages) }
+
+      before { create(:watched_item, :for_packages) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+  end
+
+  context 'when watching a request' do
+    let(:bs_request) { create(:bs_request_with_submit_action) }
+    let(:user) { create(:user) }
+
+    it 'is valid with valid attributes' do
+      expect(create(:watched_item, :for_bs_requests)).to be_valid
+    end
+    
+    context 'when missing a request' do
+      it 'is not valid' do
+        watched_item = build(:watched_item, :for_bs_requests, watchable: nil)
+        watched_item.valid?
+        expect(watched_item.errors.full_messages).to include("Watchable must exist",
+                                                            "Watchable type can't be blank",
+                                                            "Watchable can't be blank")
+      end
+    end
+    
+    context 'when missing a user' do
+      let(:watched_item) { build(:watched_item, :for_bs_requests, user: nil) }
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["User must exist"])
+      end
+    end
+
+    context 'when the user tries to add the same request twice to the watchlist' do
+      let(:bs_request) { create(:bs_request_with_submit_action) }
+      let(:watched_item) { build(:watched_item, :for_bs_requests, watchable: bs_request, user: user) }
+
+      before { create(:watched_item, :for_bs_requests, watchable: bs_request, user: user) }
+
+      it 'is not valid' do
+        expect(watched_item).not_to be_valid
+        expect(watched_item.errors.full_messages).to eql(["Watchable has already been taken"])
+      end
+    end
+
+    context 'when the user tries to add different requests to the watchlist' do
+      let(:watched_item) { build(:watched_item, :for_bs_requests, user: user) }
+
+      before { create(:watched_item, :for_bs_requests, user: user) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+
+    context 'when two different users try to watch the same package' do
+      let(:watched_item) { create(:watched_item, :for_bs_requests) }
+
+      before { create(:watched_item, :for_bs_requests) }
+
+      it 'is valid' do
+        expect(watched_item).to be_valid
+      end
+    end
+  end
```

Comment:
Same. I think specifying the same **bs_request** in both factory creations is missing here :thinking: